lib/hash.cc

```diff
@@ -268,14 +247,9 @@ hashFreeItems(hash_table * hid, HASHFREE * free_func)
     xfree(list);
 }
 
-void
-hashFreeMemory(hash_table * hid)
-{
-    if (hid == NULL)
-        return;
-    if (hid->buckets)
-        xfree(hid->buckets);
-    xfree(hid);
+hash_table::~hash_table() {
```

Comment:
Please keep the Squid coding style with brackets on a separate line.
```suggestion
hash_table::~hash_table()
{
```